File: prepare/leetcode/src/lc/s1/p01/Runner.java
package lc.s1.p01;

import static util.IO.*;

public class Runner {
    public static void main(String[] args) {
        scan(Runner.class, "in1.txt", in -> {
            int n = in.nextInt();
            int[] nums = new int[n];
            for( int i=0; i<n; i++)
                nums[i] = in.nextInt();
            int target = in.nextInt();
            new Solution().twoSum(nums, target);
            System.out.println("Hello, World");
        });
    }

}

File: prepare/leetcode/src/lc/s1/p01/Runner.py
from typing import List
import fileinput

def main():
    with open('in.txt') as f:
        lines=f.readlines()
        lines = fileinput.input('in.txt')
        n = int(lines[0])
        nums = [list(map(int, lines[1].split())) for i in range(n)]
        target = int(lines[2])
        print('two')
        print ("Solution: {} -> {}".format(nums, target))
        print('two')

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        for i in range(0, len(nums)):
            if nums[i] in map:
                return [i, map[nums[i]]]
            map[target-nums[i]] = i
        return []

if __name__ == "__main__":
    main()
